lib/slayer/form.rb

```diff
@@ -1,17 +1,31 @@
+require 'virtus'
+require 'dry-validation'
+
 module Slayer
   class Form
-    extend ModelAttribute
-    include ActiveModel::Validations if defined?(Rails)
+    attr_reader :errors
+    include Virtus.model
 
-    # Internal: Form Class Methods
     class << self
-      def from_params(params)
-        raise NotImplementedError
-      end
+      attr_accessor :_schema
 
-      def from_model(model)
-        raise NotImplementedError
+      def validations(&block)
+        self._schema = Dry::Validation.Schema(&block)
       end
     end
+
+    def validate!
+      result  = self.class._schema.call(self.attributes)
+      @errors = result.errors
+    end
+
+    def valid?
+      validate!
+      @errors.nil? || @errors.empty?
+    end
+
+    def invalid?
```

Comment:
I think that's a fair point, and I think he's right that it's an anti-pattern. Where I think I disagree --- at least a little --- is that the point of the form object is, really, to serve as a place to validate data that can be invalid, as it's input by the user. Basic ORM models, though, have always felt like a weird place to ask for validity.

I guess a way to structure this would be to have a `form` class and a `validator` class, and then you can ask the `validator` if a given `form` is valid, but to me at least it doesn't seem that weird to share that functionality in the `form` object.